packages/sdk/src/lend/providers/morpho/index.ts

```diff
@@ -53,97 +52,101 @@ export class LendProviderMorpho extends LendProvider<MorphoLendConfig> {
         )
       }
-      const depositCallData = MetaMorphoAction.deposit(amount, receiver)
+      const depositCallData = MetaMorphoAction.deposit(
+        params.amountWei,
+        receiver,
+      )
 
-      // 4. Create approval transaction data for USDC if needed
+      // Create approval transaction data if needed
       const approvalCallData = encodeFunctionData({
         abi: erc20Abi,
         functionName: 'approve',
-        args: [selectedVaultAddress, amount],
+        args: [params.marketId.address, params.amountWei],
       })
 
-      // 5. Return transaction details with real call data
+      // Return transaction details with real call data
       const currentTimestamp = Math.floor(Date.now() / 1000)
 
       return {
-        amount,
-        asset,
-        marketId: selectedVaultAddress,
+        amount: params.amountWei,
+        asset: assetAddress,
+        marketId: params.marketId.address,
         apy: vaultInfo.apy,
         timestamp: currentTimestamp,
         transactionData: {
           // Approval transaction
           approval: {
-            to: asset,
+            to: assetAddress,
             data: approvalCallData,
             value: 0n,
           },
           // Deposit transaction
           deposit: {
-            to: selectedVaultAddress,
+            to: params.marketId.address,
             data: depositCallData,
             value: 0n,
           },
         },
-        slippage: options?.slippage || this._config.defaultSlippage,
+        slippage: params.options?.slippage || this._config.defaultSlippage,
       }
     } catch (error) {
       throw new Error(
-        `Failed to lend ${amount} of ${asset}: ${
+        `Failed to open position with ${params.amountWei} of ${params.asset.metadata.symbol}: ${
           error instanceof Error ? error.message : 'Unknown error'
         }`,
       )
     }
   }
 
   /**
-   * Withdraw assets from a Morpho market
+   * Close a position in a Morpho market
    * @description Withdraws assets from a Morpho market using Blue_Withdraw operation
-   * @param params - Withdrawal operation parameters
+   * @param params - Position closing operation parameters
    * @returns Promise resolving to withdrawal transaction details
    */
-  protected async _withdraw({
-    asset,
-    amount,
-    chainId,
-    marketId,
-    options,
-  }: WithdrawParams): Promise<LendTransaction> {
+  protected async _closePosition(
+    params: LendClosePositionParams,
+  ): Promise<LendTransaction> {
     // TODO: Implement withdrawal functionality
 
-    const _unused = { asset, amount, chainId, marketId, options }
-    throw new Error('Withdraw functionality not yet implemented')
+    const _unused = {
+      asset: params.asset,
+      amount: params.amount,
+      chainId: params.chainId,
+      marketId: params.marketId,
+      options: params.options,
+    }
+    throw new Error('Close position functionality not yet implemented')
```

Comment:
Implementing in this [ticket](https://github.com/orgs/ethereum-optimism/projects/129/views/1?pane=issue&itemId=127920663&issue=ethereum-optimism%7Cverbs%7C73)